cmd/libsnap-confine-private/snap.c

```diff
@@ -48,6 +48,34 @@ bool verify_security_tag(const char *security_tag)
 	return (status == 0);
 }
 
+bool verify_hook_security_tag_name(const char *security_tag)
+{
+	const char *whitelist_re =
+	    "^snap\\.[a-z](-?[a-z0-9])*\\.(hook\\.[a-z](-?[a-z])*)$";
+
+	regex_t re;
+	if (regcomp(&re, whitelist_re, REG_EXTENDED | REG_NOSUB) != 0)
+		die("can not compile regex %s", whitelist_re);
+
+	int status = regexec(&re, security_tag, 0, NULL, 0);
+	regfree(&re);
+
+	return (status == 0);
+}
+
+bool verify_snap_name(const char *name)
```

Comment:
We already have that but it was NACKed. I think I have a better version without the regular expression anymore. Let me find that.